src/jupyter_contrib_nbextensions/nbconvert_support/embedhtml.py

```diff
@@ -49,29 +53,37 @@ def replfunc(self, match):
             with open(filename, 'rb') as f:
                 data = f.read()
 
-        self.log.info("embedding url: %s, format: %s" % (url, imgformat))
+        
         b64_data = base64.b64encode(data).decode("utf-8")
         if imgformat == "svg":
-            img = '<img src="data:image/svg+xml;base64,' + \
-                b64_data + '"'
+            prefix = "data:image/svg+xml;base64,"
         elif imgformat == "pdf":
-            img = '<img src="data:application/pdf;base64,' + \
-                b64_data + '"'
+            prefix= "data:application/pdf;base64,"
         else:
-            img = '<img src="data:image/' + imgformat + \
-                ';base64,' + b64_data + '"'
-        return img
+            prefix = "data:image/" + imgformat + ';base64,'
 
+        node.attrib["src"] = prefix + b64_data
```

Comment:
Can you preserve the original attributes of the node and only change the `src` attribute ?